allauth/socialaccount/providers/feishu/views.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+import requests
+
+from django.urls import reverse
+
+from allauth.account import app_settings
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+from allauth.socialaccount.providers.oauth2.client import (
+    OAuth2Error,
+)
+from allauth.utils import build_absolute_uri
+
+from .client import FeishuOAuth2Client
+from .provider import FeishuProvider
+
+
+class FeishuOAuth2Adapter(OAuth2Adapter):
+    provider_id = FeishuProvider.id
+
+    authorization_url = 'https://open.feishu.cn/open-apis/authen/v1/index'
+    access_token_url = 'https://open.feishu.cn/open-apis/authen/v1/access_token'
+    app_access_token_url = 'https://open.feishu.cn/open-apis/auth/v3/app_access_token/internal/'
+    user_info_url = 'https://open.feishu.cn/open-apis/authen/v1/user_info'
+
+    @property
+    def authorize_url(self):
+        settings = self.get_provider().get_settings()
+        url = settings.get(
+            "AUTHORIZE_URL", self.authorization_url
+        )
+        return url
+
+    def complete_login(self, request, app, token, **kwargs):
+        openid = kwargs.get("response", {}).get("openid")
+        resp = requests.get(
+            self.user_info_url,
+            headers={'Content-Type': 'application/json',
+                     'Authorization': f'Bearer {token}', },
+        )
+        extra_data = resp.json()
```

Comment:
Can you add a resp.raise_for_status() here?

Reply:
Have been added